crates/goose/src/providers/auto_detect.rs

```diff
@@ -0,0 +1,163 @@
+use crate::model::ModelConfig;
+
+/// Detect the provider based on API key format
+fn detect_provider_from_key_format(api_key: &str) -> Option<&'static str> {
```

Comment:
I think we discussed this; let's not do this at all. If the user enters an invalid API key we will just go on and configure the provider with it. also it is rather brittle.